src/blockdata/script.rs

```diff
@@ -49,6 +51,15 @@ use schnorr::{TapTweak, TweakedPublicKey, UntweakedPublicKey};
 #[derive(Clone, Default, PartialOrd, Ord, PartialEq, Eq, Hash)]
 pub struct Script(Box<[u8]>);
 
+impl<I: SliceIndex<[u8]>> Index<I> for Script {
```

Comment:
Wouldn't `[u8]: Index<I>` work? Looks more intuitive and general to me.

Reply:
I'm not exactly sure what you mean with this syntax suggestion, I tried
```

impl Index<usize> for Script {
    type Output = u8;

    #[inline]
    fn index(&self, index: usize) -> &Self::Output {
        &self.0[index]
    }
}
```
Is that what you meant (it doesn't work however because we need the range indexing stuff).